src/main/java/com/aws/greengrass/util/platforms/unix/UnixPlatform.java

```diff
@@ -355,20 +355,36 @@ public UnixRunWithGenerator getRunWithGenerator() {
 
     @Override
     public void createUser(String user) throws IOException {
-        runCmd("useradd -r -m " + user, o -> {
-        }, "Failed to create user");
+        try {
+            runCmd("useradd -r -m " + user, o -> {
+            }, "Failed to create user");
+        } catch (IOException e) {
+            runCmd("adduser -S  " + user, l -> {
+            }, "Failed to create user");
+        }
```

Comment:
Instead of just trying both commands, I would prefer if we can identify which command will work before calling it.